### PR TITLE
feat: add cache config options and key generation

### PR DESCRIPTION
**Pull request by chrisllontop**

### CODE REVIEW

File: src/hyphenClient.ts
Comment (osbornm) [inline src/hyphenClient.ts]: if you are going through the trouble of making specific cache.get and cache.set function beyond the defaults then you should change the signiture to match your needs. 

So the this get fuction should take the context and handle calculating the key and not make the consumer do both.

File: src/cacheClient.ts
Comment (osbornm) [inline src/cacheClient.ts:11]: this doesn't work since 0 is a valid ttl

File: src/cacheClient.ts
Comment (osbornm) [inline src/cacheClient.ts]: what happens if context is a very large object? will things overflow will it cause the cache to ballon in size? 

Is this the best way to generate a key? should we use something like https://www.npmjs.com/package/object-hash instead? ideally the way a key is calculated across all our provider languages is the same...

**Review by osbornm — CHANGES_REQUESTED** [review]

File: src/cacheClient.ts
Comment (chrisllontop) [reply src/cacheClient.ts]: My bad, this was a placeholder.

File: src/hyphenClient.ts
Comment (chrisllontop) [reply src/hyphenClient.ts]: Yep, I was doing this because I was trying to expose set and get functions as is.

File: src/cacheClient.ts
Comment (chrisllontop) [reply src/cacheClient.ts:11]: @osbornm, should we interpret `0 ` as “use the default value” or as “no expiration”?

File: src/types.ts
Comment (osbornm) [inline src/types.ts]:
```suggestion
    generateCacheKeyFn?: GenerateCacheKeyFn;
```

this is a nit pick but the existing name makes it seem like a bool value. Not everyone will be using typescript and have the types

**Review by osbornm — APPROVED** [review]
Please fix the one small naming issue. I am approving so you do not need to wait for another review.

**Pull request merged**
